python/lsst/meas/extensions/shapeHSM/_hsm_shape.py

```diff
@@ -0,0 +1,340 @@
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+import lsst.meas.base as measBase
+import lsst.pex.config as pexConfig
+from lsst.geom import Point2I
+
+__all__ = [
+    "HsmShapeBjConfig",
+    "HsmShapeBjPlugin",
+    "HsmShapeLinearConfig",
+    "HsmShapeLinearPlugin",
+    "HsmShapeKsbConfig",
+    "HsmShapeKsbPlugin",
+    "HsmShapeRegaussConfig",
+    "HsmShapeRegaussPlugin",
+]
+
+
+class HsmShapeConfig(measBase.SingleFramePluginConfig):
+    """Base configuration for HSM shape measurement."""
+
+    shearType = pexConfig.ChoiceField[str](
+        doc="The desired method of PSF correction using GalSim. The first "
+        "three options return an e-type distortion, whereas the last option returns a g-type shear.",
+        allowed={
+            "REGAUSS": "Regaussianization method from Hirata & Seljak (2003)",
+            "LINEAR": "A modification by Hirata & Seljak (2003) of methods in Bernstein & Jarvis (2002)",
+            "BJ": "From Bernstein & Jarvis (2002)",
+            "KSB": "From Kaiser, Squires, & Broadhurst (1995)",
+        },
+        default="REGAUSS",
+    )
+
+    deblendNChild = pexConfig.Field[str](
+        doc="Field name for number of deblend children.",
+        default="",  # FIXME: what's the default?
+    )
+
+    badMaskPlanes = pexConfig.ListField[str](
+        doc="Mask planes that indicate pixels that should be excluded from the fit.",
+        default=["BAD", "SAT"],
+    )
+
+
+class HsmShapePlugin(measBase.SingleFramePlugin):
+    """Base plugin for HSM shape measurement."""
+
+    ConfigClass = HsmShapeConfig
+    doc = ""
+
+    def __init__(self, config, name, schema, metadata, logName=None):
+        if logName is None:
+            logName = __name__
+        super().__init__(config, name, schema, metadata, logName=logName)
+
+        # Define flags for possible issues that might arise during measurement.
+        flagDefs = measBase.FlagDefinitionList()
+        self.FAILURE = flagDefs.addFailureFlag("General failure flag, set if anything went wrong")
+        self.NO_PIXELS = flagDefs.add("flag_no_pixels", "No pixels to measure")
+        self.NOT_CONTAINED = flagDefs.add(
+            "flag_not_contained", "Center not contained in footprint bounding box"
+        )
+        self.PARENT_SOURCE = flagDefs.add("flag_parent_source", "Parent source, ignored")
+        self.GALSIM = flagDefs.add("flag_galsim", "GalSim failure")
+
+        # Embed the flag definitions in the schema using a flag handler.
+        self.flagHandler = measBase.FlagHandler.addFields(schema, name, flagDefs)
+
+        # Utilize a safe centroid extractor that uses the detection footprint
+        # as a fallback if necessary.
+        self.centroidExtractor = measBase.SafeCentroidExtractor(schema, name)
+
+        # Define symbol for measurement type based on shearType.
+        # "g": g-type shear, "e": e-type distortion.
+        self.measTypeSymbol = "g" if config.shearType == "KSB" else "e"
```

Comment:
This is perhaps best set as a class property, where each plugin will set its own `measTypeSymbol` instead of the base class setting it.

Reply:
Do you mean something like this?
```
class HsmShapeRegaussConfig(HsmShapeConfig):
    def setDefaults(self):
        super().setDefaults()
        self.shearType = "REGAUSS"
        self.measTypeSymbol = "e"
```
Doesn't it need to exist in the base class with some default?

Reply:
`measTypeSymbol` won't be accessible from the config. It's not configurable anyway. Just the latter would be good. It's okay for the base class to not have it defined since the base class is not a registered plugin and should not be run by itself anyway.